Server/mods/deathmatch/logic/lua/CLuaTimerManager.cpp

```diff
@@ -113,6 +116,15 @@ void CLuaTimerManager::RemoveAllTimers()
     m_pProcessingTimer = NULL;
 }
 
+void CLuaTimerManager::SetTimerPaused(CLuaTimer* timer, bool paused)
+{
+    assert(timer);
+
+    timer->SetPaused(paused);
+    if (paused)
+        ListRemove(m_ProcessQueue, timer);
```

Comment:
This is not necessary because Lua is single-threaded and `SetTimerPaused` cannot be invoked in the middle of `DoPulse`. At this point `m_ProcessQueue` must be empty.

Reply:
If thats the case then the timer can be stopped from lua directly without invoking the timer manager ^

Reply:
Ah, I see. Okay then.